### ISSUE TITLE
Unable to get `text()` from a string wrapped with curly braces

### ISSUE THREAD

**Issue by DaveStein** [airbnb/enzyme#1514]
### Current behavior
I have a couple of lines like this in my test:
```
const app = shallow(<App {...state} />);
expect(app.find('#store').text()).toEqual(JSON.stringify(state));
```

The markup is this: `<script type="text/html" id="store" dangerouslySetInnerHTML={{__html: JSON.stringify(this.props)}}></script>`.

I believe it's failing because it translates to a string with curly braces. For example, `.text()` will return...

- `"Hello"` from `<script type="text/html" id="store">Hello</script>` 
- `""`from `<script type="text/html" id="store">{}</script>` 

### Expected behavior
A way to get the contents, even if they are JSON/JS.

#### API

- [X] shallow
- [ ] mount
- [ ] render

#### Version

| library         | version
| ---------------- | -------
| Enzyme           | ^3.3.0
| React            |  ^16.2

#### Adapter

- [ X] enzyme-adapter-react-16
- [ ] enzyme-adapter-react-15
- [ ] enzyme-adapter-react-15.4
- [ ] enzyme-adapter-react-14
- [ ] enzyme-adapter-react-13
- [ ] enzyme-adapter-react-helper
- [ ] others ( )

**Comment by karak** [comment]
mount(), rather than shallow(), provides what you want.

Shallow-renderer doesn't render the internal components.

Or, dive() also would help you in general cases,
but didn't in this case (I don't know why, sorry).

Try the following codes:

```js

it('test deep node', () => {
    function Script(props) {
        return <span>{props.text}</span>;
    }

    function App(props) {
        return (
            <div>
                <Script type="text/html" id="store" text={JSON.stringify(props)}></Script>
            </div>
        );
    }
    
    const state = { "hello": "HELLO" };
    
    expect(mount(<App {...state} />).find('#store').text()).toEqual(JSON.stringify(state));
    
    expect(shallow(<App {...state} />).find('#store').dive().text()).toEqual(JSON.stringify(state));

});

```

**Comment by DaveStein** [comment]
This is not an issue with shallow. As I indicated above I tried markup that does _not_ use any prop values, aside from what I really wanted with the innerHTML. The string `Hello` returns fine. The string `{}` does not. I also wouldn't think that a regular HTML node like `script` would count as a subcomponent that wouldn't get rendered with `shallow`. I had also tried using `render` with same effect.

**Comment by karak** [comment]
＠DaveStein
Oh, I confused with out past issues I remember.

However, HTML with no `{}` still returns empty as long as I tested,
and `mount` success, while `shallow` and `render` fail.

Here's my test code:

```js

function App(props) {
    return (
        <div>
            <script type="text/html" id="store" dangerouslySetInnerHTML={{__html: "HELLO"}}></script>
        </div>
    );
}

it('test render', () => {
    expect(render(<App />).find('#store').text()).not.toBe(''); // NG
});

it('test mount', () => {
    expect(mount(<App />).find('#store').text()).not.toBe(''); // OK
});

it('test shallow', () => {
    expect(shallow(<App />).find('#store').text()).not.toBe(''); // NG
});
```

**Comment by ljharb** [comment]
This is a good point; `.text()` reads out the children property, but in `shallow` in particular, it's probably not reading from dangerouslySetInnerHTML.

Separately, it's possible that `render` has a bug with `dangerouslySetInnerHTML`.

**Comment by DaveStein** [comment]
@karak in your example, did you meant to set `{}` as your innerHTML? Cause I believe `HELLO` prints fine in all cases.

**Comment by karak** [comment]
@DaveStein No, just set "HELLO" and got empty in some cases, as far as I tested, on react 16.2.0 and enzyme 3.3.0.

**Comment by karak** [comment]
@ljharb I found Cheerio 1.0.0-rc.2, depended by Enzyme 3.3.0, is going to have a breaking change by <https://github.com/cheeriojs/cheerio/issues/1050>. Is the dependency considered to be valid?

**Comment by ljharb** [comment]
`render` will do whatever `cheerio` does.
